core/src/test/java/org/elasticsearch/search/suggest/SuggestionEntryTests.java

```diff
@@ -0,0 +1,164 @@
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.Function;
+import java.util.function.Supplier;
+
+import static org.elasticsearch.common.xcontent.XContentHelper.toXContent;
+import static org.elasticsearch.common.xcontent.XContentParserUtils.ensureExpectedToken;
+import static org.elasticsearch.test.hamcrest.ElasticsearchAssertions.assertToXContentEquivalent;
+
+public class SuggestionEntryTests extends ESTestCase {
+
+    @SuppressWarnings("rawtypes")
+    private static final Map<Class<? extends Entry>, Function<XContentParser, ? extends Entry>> ENTRY_PARSERS = new HashMap<>();
+    {
+        ENTRY_PARSERS.put(Suggestion.Entry.class, Suggestion.Entry::fromXContent);
+        ENTRY_PARSERS.put(TermSuggestion.Entry.class, TermSuggestion.Entry::fromXContent);
+        ENTRY_PARSERS.put(PhraseSuggestion.Entry.class, PhraseSuggestion.Entry::fromXContent);
+        ENTRY_PARSERS.put(CompletionSuggestion.Entry.class, CompletionSuggestion.Entry::fromXContent);
+    }
+
+    /**
+     * Create a randomized Suggestion.Entry, determined by the type argument.
+     * @param type determines the entry type for this test. Valid arguments are 0 for {@link TermSuggestion.Entry},
+     * 1 for {@link PhraseSuggestion.Entry} and 2 for {@link CompletionSuggestion.Entry}
+     */
+    @SuppressWarnings("unchecked")
+    public static <O extends Option> Entry<O> createTestItem(Class<? extends Entry> entryType) {
+        Text entryText = new Text(randomAsciiOfLengthBetween(5, 15));
+        int offset = randomInt();
+        int length = randomInt();
+        @SuppressWarnings("rawtypes")
+        Entry entry = null;
+        Supplier<Option> supplier = null;
+        if (entryType == Suggestion.Entry.class) {
+            entry = new Suggestion.Entry<>(entryText, offset, length);
+            supplier = SuggestionOptionTests::createTestItem;
+        } else if (entryType == TermSuggestion.Entry.class) {
+                entry = new TermSuggestion.Entry(entryText, offset, length);
+                supplier = TermSuggestionOptionTests::createTestItem;
+        } else if (entryType == PhraseSuggestion.Entry.class) {
+                entry = new PhraseSuggestion.Entry(entryText, offset, length, randomDouble());
+                supplier = SuggestionOptionTests::createTestItem;
+        } else if (entryType == CompletionSuggestion.Entry.class) {
+                entry = new CompletionSuggestion.Entry(entryText, offset, length);
+                supplier = CompletionSuggestionOptionTests::createTestItem;
+        }
+        int numOptions = randomIntBetween(0, 5);
+        for (int i = 0; i < numOptions; i++) {
+            entry.addOption(supplier.get());
+        }
+        return entry;
+    }
+
+    @SuppressWarnings("unchecked")
+    public void testFromXContent() throws IOException {
+        for (Class<? extends Entry> entryType : ENTRY_PARSERS.keySet()) {
+        Entry<Option> entry = createTestItem(entryType);
```

Comment:
Wrong formating, or maybe you didn't want to iterate over each entry type and randomize it instead?